test/jit/test_pdt.py

```diff
@@ -65,11 +66,41 @@ def forward(self, x):
         make_global(NestedPDTInner, NestedModulePDTWrapper)
         inner_pdt_model = NestedPDTInner()
         wrapped_pdt_model = NestedModulePDTWrapper(inner_pdt_model)
-        scripted_pdt_model = torch.jit._script_pdt(wrapped_pdt_model, example_inputs=[(20, ), (2.7, ), (False, )])
+        inp: List[Tuple[Any, ...]] = [(20, ), (False, )]
+        scripted_pdt_model = torch.jit._script_pdt(wrapped_pdt_model, example_inputs={wrapped_pdt_model: inp})
         self.assertEqual(scripted_pdt_model(30), wrapped_pdt_model(30))
         self.assertEqual(scripted_pdt_model(1.9), wrapped_pdt_model(1.9))
         self.assertTrue(scripted_pdt_model(True), wrapped_pdt_model(True))
 
+    def test_nested_nn_module_class_with_args(self):
+        class NestedModulePDTInner(torch.nn.Module):
+            def __init__(self):
+                super().__init__()
+
+            def forward(self, x, y):
+                if isinstance(x, int):
+                    return x * 10 + y
+                return x
+
+        class NestedModulePDTOuter(torch.nn.Module):
+            def __init__(self, inner):
+                super().__init__()
+                self.inner = inner
+
+            def forward(self, x):
+                return self.inner(x, 20)
+
+        make_global(NestedModulePDTInner, NestedModulePDTOuter)
+        inner_pdt_model = NestedModulePDTInner()
+        outer_pdt_model = NestedModulePDTOuter(inner_pdt_model)
+        inner_input: List[Tuple[Any, ...]] = [(10, 10), (1.9, 20), ]
+        outer_input: List[Tuple[Any, ...]] = [(20, ), (False, )]
+        scripted_pdt_model = torch.jit._script_pdt(outer_pdt_model, example_inputs={inner_pdt_model: inner_input,
+                                                   outer_pdt_model: outer_input, })
```

Comment:
`inner_pdt_model: inner_input` should not be needed here. Thus we should be able to pass in `outer_input` in simple format of `List[Tuple]`. Simple format input can be used in following few test cases as well.